solr/core/src/java/org/apache/solr/schema/ManagedIndexSchemaFactory.java

```diff
@@ -223,15 +257,9 @@ private InputStream readSchemaLocally() {
     InputStream schemaInputStream = null;
     try {
       // Attempt to load the managed schema
-      try {
-        schemaInputStream = loader.openResource(managedSchemaResourceName);
-      }
-      catch (IOException e) {
-        //   Attempt to load the managed schema with the legacy name.
-        log.info("The schema is configured as managed, but managed schema resource {}  not found - looking for legacy {} instead"
-            , managedSchemaResourceName, LEGACY_MANAGED_SCHEMA_RESOURCE_NAME);
-        schemaInputStream = loader.openResource(LEGACY_MANAGED_SCHEMA_RESOURCE_NAME);
-      }
+      final String managedSchemaPath = lookupLocalManagedSchemaPath().toString();
+      schemaInputStream = loader.openResource(lookupLocalManagedSchemaPath().toString());
+      managedSchemaResourceName = managedSchemaPath.substring(managedSchemaPath.lastIndexOf("/")+1); // not loving this
```

Comment:
> // not loving this

Me neither!  Firstly, please avoid File in Java; use Path henceforth.  `lookupLocalManagedSchemaPath` ought to return a *Path* not *File*.  Path gives you methods to get the last piece via `getName(getNameCount()-1)` 

Also, I see no reason to call `lookupLocalManagedSchemaPath` twice as you have done.